storage/time-series/influxdb/storer/app/influx/index.js

```diff
@@ -0,0 +1,126 @@
+const {
+  ConfigManager: { getConfig, transformObjectKeys },
+  Logger,
+} = require('@dojot/microservice-sdk');
+const camelCase = require('lodash.camelcase');
+const flatten = require('flat');
+const InfluxState = require('./State');
+const InfluxDataWriter = require('./DataWriter');
+const InfluxOrgs = require('./Organizations');
+const InfluxMeasurement = require('./Measurements');
+
+const logger = new Logger('influxdb-storer:Influx');
+const { influx: configInflux } = getConfig('STORER');
+
+const { write: { options: influxWriteOptions } } = flatten.unflatten(configInflux);
+const configInfluxWriteOptions = influxWriteOptions ? flatten(influxWriteOptions) : {};
+const configInfluxWriteOptionsCamelCase = transformObjectKeys(configInfluxWriteOptions, camelCase);
+
+
+/**
+ * Wrapper for InfluxDB
+ */
+class InfluxDB {
+  /**
+     * @constructor
+     *
+     * @param {@dojot/microservice-sdk
+     *                  .ServiceStateManager} serviceState Manages the services' states,
+     *                                    providing health check and shutdown utilities.
+     */
+  constructor(serviceState) {
+    this.influxOrgs = new InfluxOrgs(
+      configInflux.url,
+      configInflux['default.token'],
+      configInflux['default.user'],
+      configInflux['default.password'],
+      configInflux['default.organization'],
+      configInflux['default.bucket'],
+      configInflux['retention.hrs'],
+    );
+    this.influxDataWriter = new InfluxDataWriter(
+      configInflux.url,
+      configInflux['default.token'],
+      configInflux['default.bucket'],
+      configInfluxWriteOptionsCamelCase,
+    );
+    this.influxMeasurement = new InfluxMeasurement(
+      configInflux.url,
+      configInflux['default.token'],
+      configInflux['default.bucket'],
+    );
+    this.influxState = new InfluxState(
+      configInflux.url,
+    );
+    this.serviceState = serviceState;
+    this.serviceState.registerService('influxdb');
```

Comment:
For me it is not `InfluxDB` (or `Kafka` wrapper) that depends on `serviceState`, but the opposite. 
Have you thought about inverting the dependency logic?

Have you ever thought about creating a custom `ServiceStateManager` implementation that would receive the `InfluxDB` and `Kafka` wrapper objects as _dependencies_ and the custom _impl_ knows how to access the `healthCheck` and `shutdown` methods of the dependent objects. It seems to make more sense the _responsibility_ of this object in this way.

Reply:
The idea is good, the problem is that sometimes we have to use it in class and ready and notready, as we talked ...